xonsh/tools.py

```diff
@@ -521,19 +513,16 @@ def always_false(x):
 
 def ensure_string(x):
     """Returns a string if x is not a string, and x if it already is."""
-    if isinstance(x, string_types):
-        return x
-    else:
-        return str(x)
+    return str(x)
```

Comment:
Sure, but strings are not interned by default, so I believe that `str(x) is x` happens to be an optimization in the CPython interpreter, rather than a key feature of the str constructor. So potentially running xonsh on other interpreters could change the behaviour.